models/labels/trader_frequencies/ethereum/trader_frequencies_ethereum.sql

```diff
@@ -0,0 +1,40 @@
+{{
+    config(
+        alias='trader_frequencies_ethereum',
+    )
+}}
+
+with
+ trader_frequencies as (
+    select
+        taker as address,
+        count(tx_hash) / datediff(max(block_time), min(block_time)) as trades_per_day
+    from (
+        select taker, block_time, tx_hash
+        from {{ ref('dex_aggregator_trades') }}
+        where blockchain = 'ethereum'
+        UNION ALL
+        select taker, block_time, tx_hash
+        from {{ ref('dex_trades') }}
+        where blockchain = 'ethereum'
+    )
+    group by taker
+    -- That have at least more than 1 trade
+    having datediff(max(block_time), min(block_time)) > 0
+ )
+
+select
+  array("ethereum") as blockchain,
+  address,
+  case
+    when trades_per_day >= 1 then 'Daily Trader'
+    when trades_per_day >= 0.142857142857 then 'Weekly Trader'
+    else 'Monthly Trader'
```

Comment:
What about putting a hard requirement on monthly and make the last category "infrequent" or "sparse"

Reply:
I think that's a good idea! I added two new labels: `Yearly Trader` and `Sparse Trader`.

Reply:
Nice! Note I'll will likely wind up in the early category instead of sparse!

Reply:
Do you have some wallets in mind to test each category?

Reply:
> Do you have some wallets in mind to test each category is working as expected?

Good point! I'll create a list of wallets and expected labels to test these.